src/components/Results/Results.tsx

```diff
@@ -1,42 +1,37 @@
-import type { ReactElement } from 'react';
-import { Bundle, BundleEntry, ResearchStudy } from 'fhir/r4';
+import { ReactElement } from 'react';
 import { Typography } from '@mui/material';
 
 import Study from './Study';
+import { ContactProps, SavedStudiesState, SaveStudyHandler } from './types';
+import { BundleEntry } from './types';
 
 export type ResultsProps = {
-  data: ResultsResponse;
+  entries: BundleEntry[];
+  state: SavedStudiesState;
+  handleSaveStudy: (study: BundleEntry) => SaveStudyHandler;
+  closestFacilities: ContactProps[];
 };
 
-export type ResultsResponse = {
-  results?: Bundle;
-  errors?: ErrorResponse[];
-};
-
-export type ErrorResponse = {
-  status: string;
-  response: string;
-};
-
-const Results = ({ data }: ResultsProps): ReactElement => {
-  const entries: BundleEntry[] = data?.results?.entry || [];
-  const studies = entries.filter(({ resource }) => resource?.resourceType === 'ResearchStudy');
-
-  return (
-    <>
-      <Typography fontWeight="normal" mb={2} variant="h6">
-        We found
-        <Typography color="common.blueDarker" component="span" fontWeight={700} variant="h6">
-          {` ${studies.length} `}
-        </Typography>
-        matching trials...
+const Results = ({ entries, state, handleSaveStudy, closestFacilities }: ResultsProps): ReactElement => (
+  <>
+    <Typography fontWeight="normal" mb={2} variant="h6">
+      We found
+      <Typography color="common.blueDarker" component="span" fontWeight={700} variant="h6">
+        {` ${entries.length} `}
       </Typography>
-
-      {studies.map((study: BundleEntry) => (
-        <Study key={study.resource.id} study={study.resource as ResearchStudy} />
-      ))}
-    </>
-  );
-};
+      matching trials...
+    </Typography>
+
+    {entries.map((entry: BundleEntry, index: number) => (
+      <Study
+        key={entry.resource.id}
+        entry={entry}
+        handleSaveStudy={handleSaveStudy(entry)}
+        isStudySaved={state.has(entry.resource.id)}
+        closestFacility={closestFacilities[index]}
```

Comment:
Writing a comment here for posterity more than asking for changes.

This does achieve what is needed for the ticket, but indexing through a secondary array will make it hard to keep the entries and their distances linked together for things such as sorting and filtering. That being said -- those features don't exist and I think that this makes sense given the way the Results/Study components expect a Bundle entry/ResearchStudy which are FHIR defined. I think these things will also be resolved in PDM-538 so I'm okay with this.

Reply:
For the future: I could refactor the results into some Record<string, Object> where the string is the id of a ResearchStudy and the Object has this structure: `{entry: BundleEntry, contactProps: ContactProps}` just so the ContactProps remains together with its corresponding ResearchStudy (inside the BundleEntry)